users/src/test/java/io/crate/operation/user/PrivilegeValidationTest.java

```diff
@@ -0,0 +1,334 @@
+/*
+ * This file is part of a module with proprietary Enterprise Features.
+ *
+ * Licensed to Crate.io Inc. ("Crate.io") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ *
+ * Unauthorized copying of this file, via any medium is strictly prohibited.
+ *
+ * To use this file, Crate.io must have given you permission to enable and
+ * use such Enterprise Features and you must have a valid Enterprise or
+ * Subscription Agreement with Crate.io.  If you enable or use the Enterprise
+ * Features, you represent and warrant that you have a valid Enterprise or
+ * Subscription Agreement with Crate.io.  Your use of the Enterprise Features
+ * if governed by the terms and conditions of your Enterprise or Subscription
+ * Agreement with Crate.io.
+ */
+
+package io.crate.operation.user;
+
+import com.google.common.collect.ImmutableSet;
+import io.crate.action.sql.Option;
+import io.crate.action.sql.SessionContext;
+import io.crate.analyze.ParameterContext;
+import io.crate.analyze.TableDefinitions;
+import io.crate.analyze.user.Privilege;
+import io.crate.data.Row;
+import io.crate.exceptions.PermissionDeniedException;
+import io.crate.exceptions.UnauthorizedException;
+import io.crate.metadata.TableIdent;
+import io.crate.metadata.blob.BlobSchemaInfo;
+import io.crate.sql.parser.SqlParser;
+import io.crate.test.integration.CrateDummyClusterServiceUnitTest;
+import io.crate.testing.SQLExecutor;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Collections;
+
+import static org.hamcrest.Matchers.is;
+
+public class PrivilegeValidationTest extends CrateDummyClusterServiceUnitTest {
```

Comment:
I think we should here just check that the validation happens for all statements and relations. what happens on validation is somehow irrelevant. we should rather consider checking the correct behaviour by testing the raiseMissingPrivilegeException() method.